controller/controller.go

```diff
@@ -122,3 +115,44 @@ func Process(version string) {
 	log.Debug().Str("Parsed API", parsedAPI.String()).Msg("")
 	HandleParsedFile(parsedAPI, o, dir)
 }
+
+func processEnvironmentVariables(o *lama2cmd.Opts, directory string) {
+	envMap, err := preprocess.GetL2EnvVariables(directory)
+	if err != nil {
+		log.Error().Str("Type", "Preprocess").Msg(err.Error())
+		os.Exit(0)
+	}
+	if o.Env == "" { // -e=''
+		marshalAndPrintJSON(envMap)
+	} else if o.Env != "UNSET" { // -e=any non-empty string
+		relevantEnvs := getRelevantEnvs(envMap, o)
+		marshalAndPrintJSON(relevantEnvs)
+	}
+}
+
+func marshalAndPrintJSON(data interface{}) {
+	filteredJSON, err := json.MarshalIndent(data, "", "  ")
+	if err != nil {
+		log.Error().Str("Type", "Preprocess").Msg(fmt.Sprintf("Failed to marshal JSON: %v", err))
+		os.Exit(0)
+	}
+	fmt.Println(string(filteredJSON))
+	os.Exit(0)
+}
+
+func getRelevantEnvs(envMap map[string]map[string]interface{}, o *lama2cmd.Opts) map[string]interface{} {
```

Comment:
All these env related functionalities should goto its own file.

Reply:
Got it